modules/document_repository/jsx/editCategoryForm.js

```diff
@@ -0,0 +1,191 @@
+    this.setFormData = this.setFormData.bind(this);
+    this.fetchData = this.fetchData.bind(this);
+  }
+
+  /**
+   * Called by React when the component has been rendered on the page.
+   */
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  /**
+   * Fetch data
+   * @return {Promise<void>}
+   */
+  fetchData() {
+    return fetch(this.props.dataURL, {credentials: 'same-origin'})
+      .then((resp) => resp.json())
+      .then((data) => this.setState({data: data, isLoaded: true}))
+      .catch((error) => {
+        this.setState({error: true});
+        console.error(error);
+      });
+  }
+
+  /**
+   * Renders the React component.
+   *
+   * @return {JSX} - React markup for the component
+   */
+  render() {
+    // Data loading error
+    if (this.state.error) {
+      return <h3>An error occured while loading the page.</h3>;
+    }
+    // Waiting for data to load
+    if (!this.state.isLoaded) {
+      return (<Loader/>);
+    }
+
+    let disabled = true;
+    let updateButton = null;
+    if (loris.userHasPermission('document_repository_categories')) {
+      disabled = false;
+      updateButton = <ButtonElement label="Edit Category"/>;
+    }
+
+    let errorSameParent = false;
+
+    if (
+      this.state.formData.categoryID==this.state.formData.newParentID
+      && this.state.formData.categoryID!=null
+      ) {
+      errorSameParent = true;
+    }
+
+    return (
+      <div className="row">
+        <div className="col-md-8 col-lg-7">
+          <FormElement
+            name="categoryEdit"
+            onSubmit={this.handleSubmit}
+          >
+            <h3>Change Name of a category</h3><br/>
+            <SelectElement
+              name="categoryID"
+              label="Category Name:"
+              options={this.state.data.fieldOptions.fileCategories}
+              onUserInput={this.setFormData}
+              required={true}
+              disabled={disabled}
+              hasError={false}
+              value={this.state.formData.categoryID}
+            />
+            <TextboxElement
+              name="categoryNameChange"
+              label="New Name for Category"
+              onUserInput={this.setFormData}
+              required={true}
+              disabled={disabled}
+              value={this.state.formData.categoryNameChange}
+            />
+            <SelectElement
+              name="newParentID"
+              label="New Parent:"
+              options={this.state.data.fieldOptions.fileCategories}
+              onUserInput={this.setFormData}
+              required={false}
+              disabled={disabled}
+              hasError={errorSameParent}
+              errorMessage={'Cannot be equal to itself'}
+              value={this.state.formData.newParentID}
+            />
+            {updateButton}
+          </FormElement>
+        </div>
+      </div>
+    );
+  }
+
+  /** *******************************************************************************
```

Comment:
Not sure what our standard is here, but in general I believe the React best practice is to have the render method as the last method in the class.

Reply:
I don't think we have a standard for that